sources/McQuery.Net/Internal/Data/ChallengeToken.cs

```diff
@@ -0,0 +1,42 @@
+using McQuery.Net.Exceptions;
+using McQuery.Net.Internal.Abstract;
+
+namespace McQuery.Net.Internal.Data;
+
+/// <summary>
+/// Secret value provided by Minecraft server to issue status requests.
+/// </summary>
+internal record ChallengeToken : IExpirable
+{
+    private const int AlivePeriod = 29;
+    private readonly DateTime _expiresAt = DateTime.UtcNow.AddSeconds(AlivePeriod);
+
+    /// <summary>
+    /// .ctor.
+    /// </summary>
+    /// <param name="data">Bytes that represents challenge token.</param>
+    /// <exception cref="ArgumentOutOfRangeException">
+    /// Number of bytes is incorrect.
+    /// </exception>
+    public ChallengeToken(byte[] data)
+    {
+        if (data.Length != 4)
+        {
+            throw new ArgumentOutOfRangeException(nameof(data), data, "Challenge token must have 4 bytes");
+        }
+
+        Data = data;
+    }
+
+    private byte[] Data { get; }
+    public bool IsExpired => DateTime.UtcNow >= _expiresAt;
+
+    public static implicit operator byte[](ChallengeToken token)
+    {
+        AlreadyExpiredException.ThrowIfExpired(token);
```

Comment:
Already means it tries to expire something more than once.

'ExpiredException' seems to be more appropriate name.